src/sqlfluff/rules/L052.py

```diff
@@ -79,7 +80,7 @@ def _handle_preceding_inline_comments(before_segment, anchor_segment):
                 if s.is_comment
                 and s.name != "block_comment"
                 and s.pos_marker.working_line_no
-                == anchor_segment.pos_marker.working_line_no
+                == anchor_segment.raw_segments[-1].pos_marker.working_line_no
```

Comment:
Is it possible `raw_segments` is empty and will fail with `IndexError`?

Reply:
Good question 🤔 . I'm going to have to look into that one.

Reply:
I looked into this one - we can actually be sure that `anchor_segment` is actually a `RawSegment` itself. Calling `.raw_segments` on a `RawSegment` just returns `[self]`, so this can just be reduced to `anchor_segment.pos_marker.working_line_no`. I've updated the mypy type hints accordingly and simplified the function.

Reply:
Nope - I'm wrong. `anchor_segment` isn't always a `RawSegment` (found that out in the now failing tests). However, `raw_segments` will always have length - either a segment will have children, or it will return `[self]` on `.raw_segments` if it's a `Rawsegment`.